src/prefect/core/task.py

```diff
@@ -1017,30 +1002,39 @@ def __init__(
     def __repr__(self) -> str:
         return "<Parameter: {self.name}>".format(self=self)
 
-    @property  # type: ignore
-    def name(self) -> str:  # type: ignore
-        return self._name
+    def __call__(self, flow: "Flow" = None) -> "Parameter":  # type: ignore
```

Comment:
@jlowin I assume you override this method to prevent passage of the other keyword arguments to parameters - would you mind adding a test for this?